src/main/java/com/oneops/proxy/config/ApplicationConfig.java

```diff
@@ -104,19 +104,20 @@ public KeywhizKeyStore ldapKeyStore(OneOpsConfig config, ResourceLoader loader)
 
   /** Returns the keywhiz http client */
   @Bean
-  @Lazy
   public KeywhizClient keywhizClient(
       OneOpsConfig config, @Qualifier("keywhizKeyStore") KeywhizKeyStore keywhizKeyStore)
       throws GeneralSecurityException {
-    return new KeywhizClient(config.getKeywhiz().getBaseUrl(), keywhizKeyStore);
+    OneOpsConfig.Keywhiz keywhiz = config.getKeywhiz();
+    return new KeywhizClient(/*config.getKeywhiz().getBaseUrl(),*/ keywhizKeyStore, keywhiz);
```

Comment:
Keywhiz config has the keywhiz store. So why do we need to inject both configs?

Reply:
@sureshg : config.getKeywhiz().getKeyStore() returns KeyStore not KeywhizKeyStore.

Reply:
https://github.com/oneops/secrets-proxy/blob/master/src/main/java/com/oneops/proxy/security/KeywhizKeyStore.java#L51
KeywhizKeyStore is created from that config. So you don't need to inject both